app/src/components/market/sections/market_sell/market_sell.tsx

```diff
@@ -473,7 +475,13 @@ const MarketSellWrapper: React.FC<Props> = (props: Props) => {
         />
       )}
       <StyledButtonContainer borderTop={true} marginTop={isNegativeAmountShares}>
-        <Button buttonType={ButtonType.secondaryLine} onClick={() => switchMarketTab(MarketDetailsTab.swap)}>
+        <Button
+          buttonType={ButtonType.secondaryLine}
+          onClick={() => {
+            switchMarketTab(MarketDetailsTab.swap)
+            history.replace(`/${marketMakerAddress}`)
```

Comment:
I don't think you need this history.replace on cancel buttons I tried it out and they seem to update accordingly without them probably because of those useEffect hooks with tab as trigger

Reply:
> I don't think you need this history.replace on cancel buttons I tried it out and they seem to update accordingly without them probably because of those useEffect hooks with tab as trigger

So I tested it again and this is not needed still...I explained in above comment why as well..